doc/managed_identity_command_guideline.md

```diff
@@ -0,0 +1,49 @@
+## Overview
+This document provides a common design of the CLI command interface for supporting Managed Identity in Azure CLI. New commands should follow it while existing commands can stay the same before a deprecation plan.
+
+## Command interface
+
+### Enable managed identity during resource creation
+Use `--mi-system-assigned` to enable system-assigned identity and `--mi-user-assigned` with space separated resource IDs to add user-assigned identities.
+
+```
+# <resource> can be acr, webapp, vm or any other resources that support managed identity
+az <resource> create ... --mi-system-assigned --mi-user-assigned <AzureResourceId1> <AzureResourceId2>
+```
+
+### Operate managed identity on existing resource
+Create the `identity` subgroup under the main resource command group. Support the below operations:
+
+1. Assign identities with `identity assign` command
+
+    Use `--system-assigned` to enable system assigned identity and `--user-assigned` with space separated recource ids to add user assigned identities.
+    ```
+    az <resource> identity assign ... --system-assigned --user-assigned <AzureResourceId1> <AzureResourceId2>
+    ```
+2. Remove identities with `identity remove` command
+
+    Use `--system-assigned` to remove system assigned identity and `--user-assigned` with space separated resource ids to remove user assigned identities.
+    ```
+    az <resource> identity remove ... --system-assigned --user-assigned <AzureResourceId1> <AzureResourceId2>
+    ```
+3. Show identities with `identity show` command
+
+    Use this command to show the managed identity type, tenant ids and principal ids of the system assigned identities and all user assigned identities.
+    ```
+    az <resource> identity show ...
+    ```
+4. Update identities with `identity update` command
+
+    Use different identity types to do the following:
+    - Remove all assigned identities
+        ```
+        az <resource> identity update ... --type none
+        ```
+    - Remove all user assigned identities if exist and enable system assigned identity if not enabled
+        ```
+        az <resource> identity update ... --type system_assigned
+        ```
+    - Remove system assigned identities only when both system assigned and user assigned identities exist
+        ```
+        az <resource> identity update ... --type user_assigned
+        ```
```

Comment:
Do we need to consider supporting updating the identity type to include both `system_assigned` and `user_assigned`?
Because when assign the identities, users can specify two kinds of identities at the same time.
> Use `--system-assigned` to enable system assigned identity and `--user-assigned` with space separated recource ids to add user assigned identities.
    ```
    az <resource> identity assign ... --system-assigned --user-assigned <AzureResourceId1> <AzureResourceId2>
    ```

If the two identity types in the `identity update` command are mutually exclusive, then the behavior of identity type in `identity assign` command and `identity update` command is inconsistent.
May I ask is this inconsistency in line with expectations?

Reply:
Good point about consistency. `--type` accept enums, so if service supports `system_assigned,user_assigned`, then it is supported by CLI, too.